configure.ac

```diff
@@ -255,7 +255,10 @@ AC_CHECK_TYPES([Elf32_Versym],,, [#include <elf.h>])   # for vdso_support.h
 AC_CHECK_FUNCS(sbrk)            # for tcmalloc to get memory
 AC_CHECK_FUNCS(__sbrk)          # for tcmalloc to get memory
 AC_CHECK_FUNCS(geteuid)         # for turning off services when run as root
-AC_CHECK_FUNCS(fork)            # for the pthread_atfork setup
+# for the pthread_atfork setup
+case "$host" in
+   *-linux*) AC_CHECK_FUNCS(fork);;
```

Comment:
sorry but this isn't right. Lots more OSes have fork (e.g. osx, bsd-s, qnx etc). Main question is why? I think intention you have is to disable fork testing on mingw, but why?

Reply:
> I think intention you have is to disable fork testing on mingw

You are right. And good point RE other OSes.
Just figured out how to properly check `if !MINGW` in Automake.

I've updated this check.

As to why we want to disable `HAVE_FORK` for MinGW, this was migrated from `mingw.h`, where we force `#undef HAVE_FORK`.
I figured having this hack officially documented in Automake/CMake is more visible and least surprising to developers.